sklearn/tree/tree.py

```diff
@@ -301,11 +301,13 @@ def fit(self, X, y, sample_weight=None, check_input=True,
                 sample_weight = expanded_class_weight
 
         # Set min_weight_leaf from min_weight_fraction_leaf
-        if self.min_weight_fraction_leaf != 0. and sample_weight is not None:
+        if sample_weight is None:
+            min_weight_leaf = int(ceil(self.min_weight_fraction_leaf *
+                                       n_samples))
+            min_samples_leaf = max(min_samples_leaf, min_weight_leaf)
```

Comment:
yeah, it seems like that works well as well. The max here is implicit because it's implemented in `Criterion` anyway.